src/dialect/mod.rs

```diff
@@ -590,6 +590,16 @@ pub trait Dialect: Debug + Any {
     fn supports_try_convert(&self) -> bool {
         false
     }
+
+    /// Returns true if the dialect supports `listen` clause
+    fn supports_listen_clause(&self) -> bool {
```

Comment:
```suggestion
    /// Returns true if the dialect supports the `LISTEN` statement
    fn supports_listen(&self) -> bool {
```
since its a top level statement rather than a clause. We can do the same for notify

Reply:
Good idea. Have Done! :)